From deb9de0b0f2b0fbd904940b96edfc8726d35d0fe Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 05 May 2019 21:55:25 +0300
Subject: [PATCH] [SW-0] pll: introduce HL_PLL_SKIP bypass value

The HL_PLL_SKIP is used to specify the PLL outputs that should be
skipped. That is, should be not touched during the PLLs initialization.
This will be used by the Gaudi IF/MESH W/A LBW code.

Change-Id: I7edc3060980738896f23ba4ed7911b79dd772d32
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index d62da81..2048db3 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -92,6 +92,8 @@
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
+		if (cfg->div_sel[sel] == HL_PLL_SKIP)
+			continue;
 
 		if (cfg->div_sel[sel] == HL_PLL_REF_CLK || bypass)
 			regmap_writel(HL_PLL_REF_CLK, reg);
@@ -114,7 +116,8 @@
 	u64 div_base;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
-		if (cfg->div_sel[sel] == HL_PLL_REF_CLK)
+		if (cfg->div_sel[sel] == HL_PLL_REF_CLK ||
+		    cfg->div_sel[sel] == HL_PLL_SKIP)
 			continue;
 
 		div_base = cfg->base + (sel * sizeof(u32));
diff --git a/common/pll.h b/common/pll.h
index 5f4f4cb..68bd340 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -35,6 +35,7 @@
 	HL_PLL_PLL_CLK,
 	HL_PLL_DIV_REF,
 	HL_PLL_DIV_PLL,
+	HL_PLL_SKIP,
 };
 
 enum hl_pll_div_out {
